Dockerfile

```diff
@@ -95,20 +96,20 @@ ENV ROOT_WEBAPP_REDIRECT=false
 ENV POSTGRES_JNDI_ENABLED=false
 ENV CONFIG_DIR=/opt/config
 ENV CONFIG_OVERRIDES_DIR=/opt/config_overrides
-ENV HEALTHCHECK_URL=http://localhost:8080/geoserver/web/wicket/resource/org.geoserver.web.GeoServerBasePage/img/logo.png
-
+ENV HEALTHCHECK_URL=http://localhost:8080/$WEBAPP_CONTEXT/web/wicket/resource/org.geoserver.web.GeoServerBasePage/img/logo.png
+ENV ROOT_HEALTHCHECK_URL=http://localhost:8080/web/wicket/resource/org.geoserver.web.GeoServerBasePage/img/logo.png
```

Comment:
In most files, you just have to swap `geoserver` for `$WEBAPP_CONTEXT`. Nice.

Only for the healthcheck, this extra `ROOT_HEALTHCHECK_URL` is needed in combination with an if/else in the actual health check, complicating it somewhat. I'd personally like it more if the extra env var and if/else would not be needed. I can't find a quick way to do that, though :-)

One possible idea: set `WEBAPP_CONTEXT` to an empty value if you want it in the root? For the "mv/mkdir" stuff, a `mkdir /some/dir/${WEBAPP_CONTEXT:-ROOT}/somewhere` would then be enough. Just brainstorming here.